desktop/src/main.rs

```diff
@@ -868,12 +874,58 @@ fn init() {
     tracing::subscriber::set_global_default(subscriber).expect("Couldn't set up global subscriber");
 }
 
-fn panic_hook() {
+fn panic_hook(info: &PanicInfo) {
     CALLSTACK.with(|callstack| {
         if let Some(callstack) = &*callstack.borrow() {
             callstack.avm2(|callstack| println!("AVM2 stack trace: {callstack}"))
         }
     });
+
+    // [NA] Let me just point out that PanicInfo::message() exists but isn't stable and that sucks.
+    let panic_text = info.to_string();
+    let message = if let Some(text) = panic_text.strip_prefix("panicked at '") {
+        let location = info.location().map(|l| l.to_string()).unwrap_or_default();
+        if let Some(text) = text.strip_suffix(&format!("', {location}")) {
+            text.trim()
+        } else {
+            text.trim()
+        }
+    } else {
+        panic_text.trim()
+    };
+    if rfd::MessageDialog::new()
+        .set_level(rfd::MessageLevel::Error)
+        .set_title("Ruffle")
+        .set_description(&format!(
+            "Ruffle has encountered a fatal error, this is a bug.\n\n\
+            {message}\n\n\
+            Please report this to us so that we can fix it. Thank you!\n\
+            Pressing Yes will open a browser window."
+        ))
+        .set_buttons(rfd::MessageButtons::YesNo)
+        .show()
+    {
+        let mut params = vec![];
+        params.push(("panic_text", info.to_string()));
+        params.push(("operating_system", os_info::get().to_string()));
+        let mut extra_info = vec![];
+        RENDER_INFO.with(|i| {
+            if let Some(render_info) = i.take() {
+                extra_info.push(render_info);
+            }
+        });
+        SWF_INFO.with(|i| {
+            if let Some(swf_name) = i.take() {
+                extra_info.push(format!("SWF: {swf_name}"));
+            }
+        });
+        if !extra_info.is_empty() {
+            params.push(("extra_info", extra_info.join("\n")));
+        }
+        if let Ok(url) = Url::parse_with_params("https://github.com/dinnerbone/ruffle/issues/new?assignees=&labels=bug&template=crash_report.yml", &params) {
```

Comment:
This should be ruffle-rs/ruffle 😄

Reply:
Haha yes it should! I changed it right before committing to get that link <_<